src/UserSpaceInstrumentation/ExecuteInProcess.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "UserSpaceInstrumentation/ExecuteInProcess.h"
+
+#include <absl/base/casts.h>
+
+#include "AllocateInTracee.h"
+#include "ExecuteMachineCode.h"
+#include "MachineCode.h"
+#include "UserSpaceInstrumentation/InjectLibraryInTracee.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace {
+
+// Size of the small amount of memory we need in the tracee to write machine code into.
+constexpr uint64_t kCodeScratchPadSize = 1024;
+
+}  // namespace
+
+ErrorMessageOr<uint64_t> ExecuteInProcess(pid_t pid, void* handle, std::string_view function,
```

Comment:
Yes you are right, currently I only need to use it once and for my use case the initial implementation would be sufficient. The comment was for a hypothetic use case where one would want to call the same function multiple times.

Reply:
Actually, I say that, but I lie. I need to call the initialize function multiple times, once for each occurence of the function pointer table (it could be that different modules have their own instance of the table).

Reply:
Okay, let's keep it as it is now (with the two functions as suggested above).